lib/src/commands/message_command.dart

```diff
@@ -0,0 +1,88 @@
+import 'dart:async';
+
+import 'package:nyxx_commands/src/checks/checks.dart';
+import 'package:nyxx_commands/src/commands/command.dart';
+import 'package:nyxx_commands/src/context/context.dart';
+import 'package:nyxx_commands/src/context/message_context.dart';
+import 'package:nyxx_commands/src/errors.dart';
+
+abstract class MessageCommand implements Command {
+  @override
+  Function(MessageContext) get execute;
+
+  factory MessageCommand(
+    String name,
+    Function(MessageContext) execute, {
+    Iterable<AbstractCheck> checks = const [],
+  }) =>
+      MessageCommandImpl(name, execute, checks: checks);
+}
+
+class MessageCommandImpl implements MessageCommand {
```

Comment:
Thats very bad naming. Please rename `MessageCommand` to `IMessageCommand` and this class to `MessageCommand`